### PR TITLE
adding the impedance_check feature

### PR DESCRIPTION
**Pull request by RobinGuillard**
Here are my proposed set of impedance checking functions!
The "Impedance_thresholding" function can be used directly from the data. It returns a list of list that should be used for the other functions.
The check_RMS function should be used between the end of the RMS function and the input of the classification and finally the "create_bad_epoch_annotation" can be used as a complement for annotating the data.

### CODE REVIEW

File: tinnsleep/check_impedance.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+           Returns
+           -------
+           annotations : list of dictionaries
+           Chronological list of bad epoch annotations
+           """
+    annotations = []
+    for i in range(len(check_imp)):
+        if np.mean(check_imp[i]) == 1:
+            annotations.append({'onset': i * interval, 'duration': duration, 'description': "1", 'orig_time': orig_time})
+    return annotations
```
Comment (lkorczowski) [inline tinnsleep/check_impedance.py:131]: As the scheme is with fixed interval, I advice using the `_sliding` keyword in such fun. Not important for now but nicer for consistency.

File: tinnsleep/check_impedance.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+           -------
+           annotations : list of dictionaries
+           Chronological list of bad epoch annotations
+           """
+    labels = []
+    for i in range(len(check_imp)):
+        if np.mean(check_imp[i]) == 1:
+            labels.append(True)
+        else:
+            labels.append(False)
```
Comment (lkorczowski) [inline tinnsleep/check_impedance.py:155]: loops !!!

File: tests/test_check_impedance.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    for i in range(200):
+        data[0][i] += 100
+        data[1][i] += 100
+    ch_names = ['1_imp', '2_imp']
+    duration = 50
+    interval = 50
+    THR = 20.0
+    check_imp = Impedance_thresholding(data, ch_names, duration, interval, THR, ch_types=['emg'])
+    npt.assert_equal(check_imp, [[True, True], [True, True], [True, True], [True, True], [False, False], [False, False],
+                                 [False, False], [False, False]])
```
Comment (lkorczowski) [inline tests/test_check_impedance.py:25]: please add cross-impedance check
```
def test_Impedance_thresholding():
    np.random.seed(42)
    data = np.random.randn(2, 400)
    for i in range(200):
        data[0][i] += 100

    for i in range(200):
        data[1][i] += 100
    ch_names = ['1_imp', '2_imp']
    duration = 50
    interval = 50
    THR = 20.0
    check_imp = Impedance_thresholding(data, ch_names, duration, interval, THR, ch_types=['emg'])
    npt.assert_equal(check_imp, [[True, True], [True, True], [True, True], [True, True], [False, False], [False, False],
                                 [False, False], [False, False]])
```

File: tinnsleep/check_impedance.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        is plotted. If None (default), it will return all channel types present. If a list of ch_types is provided,
+        it will return multiple figures.
+
+    Returns
+    -------
+    check_imp list of list of 0s and 1s shape (nb_epochs, nb_electrodes) 0s marking bad channels for the designated
+    epoch
+    """
+    raw = CreateRaw(data, imp_chan, montage=montage, ch_types=ch_types)
+    epochs = RawToEpochs_sliding(raw, duration, interval, picks=None)
```
[the remark this answers is not in the document]
Comment (RobinGuillard) [reply tinnsleep/check_impedance.py:39]: Very good idea!!

File: tinnsleep/check_impedance.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    mean_imp = np.mean(abs(epochs), axis=-1)
+    check_imp = []
+    for elm in mean_imp:
+        small_check = []
+        for chan in elm:
+            if chan > THR:
+                small_check.append(True)
+            else:
+                small_check.append(False)
+        check_imp.append(small_check)
```
[the remark this answers is not in the document]
Comment (RobinGuillard) [reply tinnsleep/check_impedance.py:47]: done with np.where()

File: tinnsleep/check_impedance.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            bad_chan_ind = []
+            mod_X.append(X[i])
+            for j in range(len(X[0])):
+                if check_imp[i][j] == 0:
+                    remaining_chan.append(X[i][j])
+                else:
+                    bad_chan_ind.append(j)
+            rem_chan_av = np.mean(remaining_chan)
+            for ind in bad_chan_ind:
+                mod_X[-1][ind] = rem_chan_av
```
[the remark this answers is not in the document]
Comment (RobinGuillard) [reply tinnsleep/check_impedance.py:84]: Franckly it would be hard, please let it be :S ?

File: tinnsleep/check_impedance.py (hunk 2 above)
Comment (RobinGuillard) [reply tinnsleep/check_impedance.py:153]: Corrected

File: tinnsleep/check_impedance.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+            Returns
+            -------
+            mod_labels : ndarray, shape (n_trials,)
+            Modified labels to fit a length of n_trials
+            """
+    labels = labels.tolist()
+    for i in range(len(check_imp)):
+        if np.mean(check_imp[i]) == 1:
+            labels.insert(i, False)
```
[the remark this answers is not in the document]
Comment (RobinGuillard) [reply tinnsleep/check_impedance.py:104]: Hardly editable. Sorry!

**Pull request merged**